src/modules/emote_menu/components/Menu.jsx

```diff
@@ -0,0 +1,143 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import Divider from 'rsuite/lib/Divider/index.js';
+import keycodes from '../../../utils/keycodes.js';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/menu.module.css';
+import Emotes from './Emotes.jsx';
+import Header from './Header.jsx';
+import Preview from './Preview.jsx';
+import Sidebar from './Sidebar.jsx';
+
+let altKeyPressed = false;
+let shiftKeyPressed = false;
+
+export default function EmoteMenu({triggerRef, appendToChat}) {
+  const [search, setSearch] = useState('');
+  const [selected, setSelected] = useState(null);
+  const [, setUpdated] = useState(false);
+
+  const [section, setSection] = useState({
+    eventKey: null,
+    scrollTo: false,
+  });
+
+  const [arrowKeys, setArrowKeys] = useState({
+    top: false,
+    down: false,
+    right: false,
+    left: false,
+  });
+
+  const onHide = useCallback(() => triggerRef.current.close(), [triggerRef]);
+
+  const handleClick = useCallback((emote) => {
+    if (altKeyPressed) {
+      emoteStore.toggleFavorite(emote);
+      return;
+    }
+
+    appendToChat(emote.code);
+    emoteStore.trackHistory(emote);
+
+    if (shiftKeyPressed) {
+      return;
+    }
+
+    onHide();
+  }, []);
+
+  useEffect(() => setSearch(''), [section]);
+
+  useEffect(() => {
+    function buttonPressCallback(event) {
+      altKeyPressed = event.altKey;
+      shiftKeyPressed = event.shiftKey;
+    }
+
+    function arrowPressCallback(event) {
+      if (
+        event.keyCode === keycodes.UpArrow ||
+        event.keyCode === keycodes.DownArrow ||
+        event.keyCode === keycodes.RightArrow ||
+        event.keyCode === keycodes.LeftArrow
+      ) {
+        setArrowKeys({
+          top: event.keyCode === keycodes.UpArrow,
```

Comment:
these do not appear to ever get set back to false since we only listen to keydown. listening to key down is fine for all of this, but we likely should be handling button presses directly for keyboard navigation. for example, a user holding down an arrow key will generate multiple arrow press events but we are already set to `true`. if we're setting state we miss that nuance.